src/test/java/pl/byczazagroda/trackexpensesappbackend/service/WalletServiceImplTest.java

```diff
@@ -145,5 +155,47 @@ void shouldThrowAnExceptionWhenNameContainsIllegalLetters() throws Exception {
 //        then
         Assertions.assertThrows(ConstraintViolationException.class, () -> walletService.createWallet(createWalletDTO));
     }
+
+    @Test
+    void shouldReturnListOfWalletDTO() {
+        // given
+        List<Wallet> walletList = createListOfWallets();
+
+        when(walletRepository.findAll()).thenReturn(walletList);
+        when(walletModelMapper.mapWalletEntityToWalletDTO(any())).thenReturn(walletModelMapper.mapWalletEntityToWalletDTO(any()));
+
+        // when
+        List<WalletDTO> allWallets = walletService.getAllWallets();
+
+        // then
+        assertThat(allWallets, hasSize(3));
+        assertThat(allWallets.get(0).name()).isEqualTo(NAME_OF_WALLET);
+        assertThat(allWallets.get(0).creationDate()).isEqualTo(walletList.get(0).getCreationDate());
+        assertThat(allWallets.get(1).name()).isEqualTo(NAME_OF_WALLET_1);
+        assertThat(allWallets.get(1).creationDate()).isEqualTo(walletList.get(1).getCreationDate());
+        assertThat(allWallets.get(2).name()).isEqualTo(NAME_OF_WALLET_2);
+        assertThat(allWallets.get(2).creationDate()).isEqualTo(walletList.get(2).getCreationDate());
+    }
+
+    @Test
+    void shouldThrowExceptionWhenListOfWalletsNotFound() {
+        // given
+        Mockito.when(walletRepository.findAll()).thenThrow(RuntimeException.class);
+
+        // when
+        Exception exception = assertThrows(RuntimeException.class, () -> walletService.getAllWallets());
+
+        // then
+        assertThat(exception)
+                .isInstanceOf(ResourceNotFoundException.class)
+                .hasMessage("An error occurred while retrieving the list of wallets");
```

Comment:
What if the class `WalletServiceImpl` is changed? If the `message` value changes? Then the test, although correct, will not work properly.
If you check the value of message in the test, you have to check this value with the value obtained from the source class of the code (`WalletServiceImpl`). In that case, in that class it has to be constant.

Reply:
- [ ] add constant value of exception message

Reply:
Constant String representing exeption message for `getWallets()` method has been added to `WalletServiceImplTest.java`